src/proposals/20250109/SparkEthereum_20250109.t.sol

```diff
@@ -0,0 +1,926 @@
+
+        assertEq(usdc.balanceOf(Ethereum.ALM_PROXY),  usdcAmount);
+        assertEq(ausdc.balanceOf(Ethereum.ALM_PROXY), 0);
+    }
+
+    function test_ETHEREUM_AaveRateLimits() public {
+        executeAllPayloadsAndBridges();
+
+        MainnetController controller = MainnetController(NEW_ALM_CONTROLLER);
+        IRateLimits rateLimits       = IRateLimits(Ethereum.ALM_RATE_LIMITS);
+        
+        IERC20 usds  = IERC20(Ethereum.USDS);
+        IERC20 usdc  = IERC20(Ethereum.USDC);
+        IERC20 ausds = IERC20(ATOKEN_USDS);
+        IERC20 ausdc = IERC20(ATOKEN_USDC);
+
+        deal(Ethereum.USDS, Ethereum.ALM_PROXY, 50_000_000e18);
+        // Use deal2 for USDC because storage is not set in a common way
+        deal2(Ethereum.USDC, Ethereum.ALM_PROXY, 50_000_000e6);
+
+        // USDS
+
+        vm.startPrank(Ethereum.ALM_RELAYER);
+
+        bytes32 usdsDepositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_DEPOSIT(),
+            address(ausds)
+        );
+        bytes32 usdsWithdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_WITHDRAW(),
+            address(ausds)
+        );
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsDepositKey), 50_000_000e18);
+        assertEq(usds.balanceOf(Ethereum.ALM_PROXY),             50_000_000e18);
+        assertEq(ausds.balanceOf(Ethereum.ALM_PROXY),            0);
+
+        controller.depositAave(address(ausds), 50_000_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsDepositKey), 0);
+        assertEq(usds.balanceOf(Ethereum.ALM_PROXY),             0);
+        assertEq(ausds.balanceOf(Ethereum.ALM_PROXY),            50_000_000e18);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsWithdrawKey), type(uint256).max);
+
+        // USDC
+
+        bytes32 usdcDepositKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_DEPOSIT(),
+            address(ausdc)
+        );
+        bytes32 usdcWithdrawKey = RateLimitHelpers.makeAssetKey(
+            controller.LIMIT_AAVE_WITHDRAW(),
+            address(ausdc)
+        );
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 50_000_000e6);
+        assertEq(usdc.balanceOf(Ethereum.ALM_PROXY),             50_000_000e6);
+        assertEq(ausdc.balanceOf(Ethereum.ALM_PROXY),            0);
+
+        controller.depositAave(address(ausdc), 50_000_000e6);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 0);
+        assertEq(usdc.balanceOf(Ethereum.ALM_PROXY),             0);
+        assertEq(ausdc.balanceOf(Ethereum.ALM_PROXY),            50_000_000e6);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdcWithdrawKey), type(uint256).max);
+
+        // All limits should be reset in 2 days + 1 (rounding)
+        skip(2 days + 1);
+
+        assertEq(rateLimits.getCurrentRateLimit(usdsDepositKey), 50_000_000e18);
+        assertEq(rateLimits.getCurrentRateLimit(usdcDepositKey), 50_000_000e6);
```

Comment:
We should update this to:

Warp an hour to show that everything is increasing at the expected rate
Warp two days to show that everything hits the expected max amounts

In theory the slope could be 50m/day and still pass here for example

Reply:
updated